CameraBrowser/CameraBrowserSample/ViewController.cs

```diff
@@ -0,0 +1,238 @@
+
+		public override void ViewDidLoad () //application did finish launching
+		{
+			base.ViewDidLoad ();
+
+			// Do any additional setup after loading the view.
+
+			CamerasController.SelectsInsertedObjects = false;
+
+			NSNotificationCenter.DefaultCenter.AddObserver(this, new Selector("processICLaunchParams:"), (NSString)"ICLaunchParamsNotification", null);
+
+			CamerasController.SelectsInsertedObjects = false;
+
+			MediaFilesController.AddObserver(this, (NSString)"selectedObjects", (NSKeyValueObservingOptions)0, IntPtr.Zero);
+
+			DeviceBrowser.Delegate = this;
+			DeviceBrowser.BrowsedDeviceTypeMask = ICBrowsedDeviceType.Local | ICBrowsedDeviceType.Remote | ICBrowsedDeviceType.Camera;
+			DeviceBrowser.Start();
+		}
+
+		[Export("observeValueForKeyPath:ofObject:change:context:")]
+		private void ObserveValueForKeyPath (NSString keyPath, NSArrayController ofObject, NSDictionary change, IntPtr context)
+		{
+			if (keyPath == "selectedObjects" && ofObject == MediaFilesController)
+			{
+				WillChangeValue("canDelete");
+				WillChangeValue("canDownload");
+				DidChangeValue("canDelete");
+				DidChangeValue("canDownload");
+
+			}
+		}
+
+		[Export("canDelete")]
+		public bool CanDelete ()
+		{
+			bool can = false;
+			InvokeOnMainThread(() => {
+				var selectedFiles = MediaFilesController.SelectedObjects;
+				foreach (ICCameraFile f in selectedFiles)
+				{
+					if (!f.Locked)
+					{
+						can = true;
+						break;
+					}
+				}
+			});
+			return can;
+		}
+
+		[Export("canDownload")]
+		public bool CanDownload()
+		{
+			var res = false;
+			InvokeOnMainThread(() => {
+				res = MediaFilesController.SelectedObjects is not null ? MediaFilesController.SelectedObjects.Length > 0 : false;
+			} );
+			return res;
+		}
+
+		[Export("selectedCamera")]
+		public ICCameraDevice? SelectedCamera
+		{
+			get
+			{
+				var selectedObjects = CamerasController.SelectedObjects;
+				if (selectedObjects is null || selectedObjects.Length == 0)
+					return null;
+				return selectedObjects[0] as ICCameraDevice;
+			}
+
+		}
+
+		[Export("downloadFiles:")]
+		public void downloadFiles (ICCameraFile[] files)
+		{
+			NSString filePath = new NSString("~/Pictures");
+			var filePath1 = filePath.ExpandTildeInPath();
+			string[] fileUrl = new string[] { filePath1 };
+
+			var objects = new[] { NSUrl.CreateFileUrl(fileUrl) };
+			var keys = new[] { ICCameraDownloadOptionKeys.DownloadsDirectoryUrl };
+			var dict = new NSDictionary<NSString, NSObject>(keys, objects);
+			foreach (ICCameraFile f in files)
+			{
+				f?.Device?.RequestDownloadFile(f, dict, didDownloadFileDel);
+			}
+		}
+
+		public void didDownloadFileDel(ICCameraFile f, NSError err, NSDictionary options)
+		{
+			Console.WriteLine("didDownloadFile called with:");
+			Console.WriteLine($"file: {f}");
+			Console.WriteLine($"error: {err}");
+			Console.WriteLine($"options: {options}");
+		}
+
+		[Export("readFiles:")]
+		public void readFiles (ICCameraFile[] files)
```

Comment:
Casing:

```suggestion
		public void ReadFiles (ICCameraFile[] files)
```

(there are more further down in this file too)